lib/Frontend/ModuleInterfaceBuilder.cpp

```diff
@@ -81,12 +83,16 @@ struct ErrorDowngradeConsumerRAII: DiagnosticConsumer {
       Diag.addConsumer(*consumer);
     }
     Diag.removeConsumer(*this);
+    if (Silence)
+      Diag.resetHadAnyError();
```

Comment:
Maybe I'm missing something, but wouldn't this cause us to ignore previous (unrelated) errors for the purposes of checking whether an error has occurred?  Maybe we can do a save/restore of the error state instead?  Or if this cannot happen, maybe we could assert there haven't been any errors yet?